server/src/main/java/io/crate/execution/dml/upsert/ShardUpsertRequest.java

```diff
@@ -51,6 +51,12 @@
 
 public final class ShardUpsertRequest extends ShardRequest<ShardUpsertRequest, ShardUpsertRequest.Item> {
 
+    public enum Operation {
```

Comment:
An Upsert Request can have 3 different semantics, INSERT, UPDATE and INSERT_UPDATE_ON_CONFLICT. The INSERT_UPDATE_ON_CONFLICT is left open at the moment.